R/filter.R

```diff
@@ -0,0 +1,111 @@
+#' Filter data by levels of numeric variables
+#'
+#' Filter a dataframe by some boolean condition of one numeric variable
+#' and returns the result along with tidyverse code used to generate it.
+#'
+#' @param data a dataframe or survey design object to filter
+#' @param var character of the column in `data` to filter by
+#' @param op  a logical operator of "<=", "<", ">=", ">", "==" or "!="
+#'        for the boolean condition
+#' @param num a number for which the \code{op} applies to
+#' @return filtered dataframe with tidyverse code attached
+#' @seealso \code{\link{code}}
+#' @rdname filter_num
+#' @examples
+#' filtered <- filter_num(iris, var = "Sepal.Length", op = "<=", num = 5)
+#' cat(code(filtered))
+#' head(filtered)
+#'
+#' library(survey)
+#' data(api)
+#' svy <- svydesign(~ dnum + snum, weights = ~pw, fpc = ~ fpc1 + fpc2, data = apiclus2)
+#' svy_filtered <- filter_num(svy, var = "api00", op = "<", num = 700)
+#' cat(code(svy_filtered))
+#'
+#' @author Owen Jin, Tom Elliott, Stephen Su
+#' @export
+#' @md
+filter_num <- function(data, var,
+                       op = c("<=", "<", ">=", ">", "==", "!="),
+                       num) {
+    op <- rlang::arg_match(op)
+    expr <- rlang::enexpr(data)
+    ## Defuse {`op`(var, num)} into the form of {var `op` num}
+    filter_expr <- rlang::expr((!!op)(!!rlang::sym(var), !!num))
+    if (is_survey(data)) {
+        if (!inherits(data, "tbl_svy")) {
+            expr <- rlang::expr(!!expr %>% srvyr::as_survey())
+        }
+        expr <- rlang::expr(!!expr %>% srvyr::filter(!!filter_expr))
+    } else {
+        expr <- rlang::expr(!!expr %>% dplyr::filter(!!filter_expr))
+    }
+    eval_code(expr)
+}
+
+
+#' Filter data by levels of categorical variables
+#'
+#' Filter a dataframe by some levels of one categorical variable
+#' and returns the result along with tidyverse code used to generate it.
+#'
+#' @param data a dataframe or survey design object to filter
+#' @param var character of the column in \code{data} to filter by
+#' @param levels a character vector of levels in \code{var} to filter by
+#'
+#' @return filtered dataframe with tidyverse code attached
+#' @seealso \code{\link{code}}
+#' @rdname filter_cat
+#' @examples
+#' filtered <- filter_cat(iris,
+#'     var = "Species",
+#'     levels = c("versicolor", "virginica")
+#' )
+#' cat(code(filtered))
+#' head(filtered)
+#'
+#' @author Owen Jin, Stephen Su
+#' @export
+filter_cat <- function(data, var, levels) {
+    expr <- rlang::enexpr(data)
+    lvls <- rlang::enexpr(levels)
+    op <- ifelse(length(levels) > 1, "%in%", "==")
+    filter_expr <- rlang::expr((!!op)(!!rlang::sym(var), !!lvls)) ## Defuse `op`
+    if (is_survey(data)) {
```

Comment:
This looks much better - I hope you agree! 
It might be handy to create a helper function here, expr <- as_svy_expr(expr, data), that either returns expr or Line 76 - that way you can just call this without testing for inheritance in every function.